addons/storyquest_bootstrap/plugin.gd

```diff
@@ -0,0 +1,165 @@
+func _close_dialogue() -> void:
+	_storyquest_dialogue.queue_free()
+	_storyquest_dialogue = null
+
+
+func _on_create_storyquest(title: String, description: String, filename: String) -> void:
+	_close_dialogue()
+
+	var storyquest_path := STORYQUESTS_PATH + filename
+	assert(not FileAccess.file_exists(storyquest_path))
+
+	var error := DirAccess.make_dir_absolute(storyquest_path)
+	assert(error == OK)
+
+	# These need to be created before the following for loop, because the intro (for instance)
+	# needs to replace the outro UID in its content.
+	var storyquest_uids := {
+		"intro": ResourceUID.id_to_text(ResourceUID.create_id()),
+		"outro": ResourceUID.id_to_text(ResourceUID.create_id()),
+	}
+
+	for i in template_info:
+		var directory: String = template_info[i]["components_path"]
+		error = DirAccess.make_dir_recursive_absolute(
+			storyquest_path + directory.replacen("template_", "")
+		)
+		assert(error == OK)
+
+		var scene_path: String = template_info[i]["scene_path"].replacen("template_", "")
+		var dialogue_path: String = template_info[i]["dialogue_path"].replacen("template_", "")
+
+		var template_dialogue: DialogueResource = ResourceLoader.load(
+			TEMPLATE_PATH + template_info[i]["dialogue_path"]
+		)
+
+		# Saving the raw text because ResourceSaver.save() fails with ERR_FILE_UNRECOGNIZED
+		var dialogue_file: FileAccess = FileAccess.open(
+			storyquest_path + dialogue_path, FileAccess.WRITE
+		)
+		dialogue_file.store_string(template_dialogue.raw_text)
+		dialogue_file.close()
+
+		var template_dialogue_uid := ResourceUID.id_to_text(
+			ResourceLoader.get_resource_uid(TEMPLATE_PATH + template_info[i]["dialogue_path"])
+		)
+
+		var template_scene_uid := ResourceUID.id_to_text(
+			ResourceLoader.get_resource_uid(TEMPLATE_PATH + template_info[i]["scene_path"])
+		)
+		var template_image_uid: String
+		var image_path: String
+		var template_image_path = template_info[i].get("image_path")
+		if template_image_path:
+			template_image_uid = ResourceUID.id_to_text(
+				ResourceLoader.get_resource_uid(TEMPLATE_PATH + template_image_path)
+			)
+			# If we use ResourceLoader.load() it brings an CompressedTexture2D from a ctex file.
+			# so if we then use resource.duplicate() and try to save it with ResourceSaver.save()
+			# to a PNG file, it fails with ERR_FILE_UNRECOGNIZED.
+			image_path = storyquest_path + "/" + template_image_path.replacen("template_", "")
+			error = DirAccess.copy_absolute(TEMPLATE_PATH + template_image_path, image_path)
+			assert(error == OK)
+
+		# The following should be enough instead of a full scan, but it doesn't work:
+		# EditorInterface.get_resource_filesystem().update_file(storyquest_path + dialogue_path)
+		# EditorInterface.get_resource_filesystem().reimport_files(
+		# 	[storyquest_path + dialogue_path]
+		# )
+		EditorInterface.get_resource_filesystem().scan()
+		if EditorInterface.get_resource_filesystem().is_scanning():
+			await EditorInterface.get_resource_filesystem().resources_reimported
```

Comment:
Is it guaranteed that this is emitted once for the whole set of files? I guess not but perhaps in practice it is enough.